### PR TITLE
Redownload of read-only file fix

### PR DESCRIPTION
**Pull request by vibhansa-msft**
- If a file is readonly and file-cache decides to reload that file then it fails because read-only file can not be written
- fix for this is to change the mode and download and post download reset the mode

### CODE REVIEW

File: component/file_cache/file_cache.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ type FileCache struct {
 
 	createEmptyFile bool
 	allowNonEmpty   bool
-	cacheTimeout    uint32
+	cacheTimeout    float64
```
Comment (gapra-msft) [inline component/file_cache/file_cache.go:66]: Why are we making this a float?

File: component/file_cache/file_cache.go [hunk 2]
```diff
@@ -697,8 +699,33 @@ func (fc *FileCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Hand
 		// If the lock is held then we cannot update the file
 		f, err = os.OpenFile(localPath, os.O_WRONLY, options.Mode)
 		if err != nil {
-			log.Err("FileCache::OpenFile : Failed to open file %s in WR mode", localPath)
-			return nil, err
+			if os.IsPermission(err) {
```
Comment (gapra-msft) [inline component/file_cache/file_cache.go:702]: You mentioned we should figure out a more efficient way to deal with this problem right? Could you create an item for it?

**Review by tasherif-msft — APPROVED** [review]

File: component/file_cache/file_cache.go (hunk 1 above)
Comment (vibhansa-msft) [reply component/file_cache/file_cache.go:66]: the time comparison part needs float so if we do not modify the base type here, every comparison will need type conversion.

File: component/file_cache/file_cache.go (hunk 2 above)
Comment (vibhansa-msft) [reply component/file_cache/file_cache.go:702]: https://msazure.visualstudio.com/One/_backlogs/backlog/XStore-DevExp/Epics/?showParents=true&workitem=13735363

**Review by souravgupta-msft — APPROVED** [review]

**Pull request merged**